### DISCUSSION TITLE
self-host static hosting bsky/ATP?

### DISCUSSION THREAD

**Discussion by est** [bluesky-social/atproto#2300] [General Discussion]
Is it possible to self-host a bsky/ATP instance with static site generator? This could save lots of cost, adoption would be faster & easier.

I understand there're limitations like you can't interact with such instance but I guess it could be still pretty useful in scenarios like announcements. Big brands disable comments so the posts are in read-only broadcasting fashion anyway.

Could this be done with current version of ATP? At least in some handicapped way?

**Comment by bnewbold** [comment]
I'm not sure it is impossible, but it would be pretty hard right now. The Relay expects to consume from a WebSocket.

We have thought about this for a long time, and would like a way to post a simple .car file somewhere containing the repository, and have the user's DID document point to that file. Relays would periodically check if the file has changed, and site-generators could poke a configured set of Relays to have them update.

This would be some large-ish changes to what Relays need to do, and to make it work effectively we'd basically have to specify that all Relays need to support this functionality (not be an optional thing which only some Relays support).

So, it is on the radar, and we want to do it at some point, but no timeline.

**Comment by est** [comment]
@bnewbold 

So here's an idea, I  static-host a series of `Record`s and `Blob`s on Github Pages (and alikes), then I write another "relay" that speaks Websocket or XRPC via Cloudflare Worker. Does this work? Any suggestions?

**Reply by snarfed** [reply]
Sounds like a PDS (not relay) that users can bring their own repo storage for, eg with static CAR files. The PDS would need to check that users' CAR files and blocks, commits, etc are valid, but apart from that, this should be doable!

**Reply by hiinaspace** [reply]
I had this same idea and made a prototype: https://github.com/hiinaspace/satrepo . It seems to work well enough. There might still be weird limitations I haven't thought about (in addition to the limitations that I know about but still are unimplemented). But I'm feeling relatively confident about the overall shape.
